c7n/schema.py

```diff
@@ -258,7 +258,7 @@ def generate(resource_types=()):
                 'comments': {'type': 'string'},
                 'description': {'type': 'string'},
                 'tags': {'type': 'array', 'items': {'type': 'string'}},
-                'metadata': {'$ref': '#/definitions/basic_dict'},
+                'metadata': {'type': ['object', 'array']},
```

Comment:
just object, people can put an array under a key, also jsonschema that would be a oneOf / anyOf block to support both afaik

Reply:
Ok cool, sounds good. Just made the update in 9753620

Reply:
Heads up, I also needed to re-record an IAM test in 38ddfaa to pass CI. Kind of weird it randomly stopped working but seems to be fine now 🤷.

Reply:
Hmm, actually that test seems to be a timebomb that will break every 30 days. Rethinking what the fix should be 🤔

Reply:
Ok, d044909 should do it.

Reply:
thanks! re flakey